vmcnet/physics/potential.py

```diff
@@ -146,3 +148,40 @@ def potential_fn(params: ModelParams, x: Array) -> Array:
         return constant_potential
 
     return potential_fn
+
+
+
+def create_hubbard_potential(
```

Comment:
In terms of this function, if we're using integer indices as input, would making use of something like `jax.numpy.intersect1d` make more sense? Can rely on the nice algorithm implemented there.